### DISCUSSION TITLE
errors with brew update: fatal: Could not resolve HEAD to a revision. brew doctor errors

### DISCUSSION THREAD

**Discussion by lsaffel** [Homebrew/discussions#1512] [Getting started]
Hi. I have Mac OS Big Sur, the latest, 11.3.1
LV-Mac-Mini-64:~ Lisa$ `brew update`
fatal: Could not resolve HEAD to a revision
Already up-to-date.
..............................................................
So, I tried brew doctor
..............................................................
LV-Mac-Mini-64:~ Lisa$ `brew doctor`
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8-config

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libpolar-3.6.0.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/node/cppgc/allocation.h
  /usr/local/include/node/cppgc/common.h
  /usr/local/include/node/cppgc/custom-space.h
..............................................................
(a long, long list of files continues after this, screens of them)

Any suggestions on how to fix this ? Thanks.

**Comment by carlocab** [comment]
Anything else aside from the list of files from `brew doctor`? What's the output of `brew config`?

**Reply by lsaffel** [reply]
at the end of that long list of files, it says this:

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
LV-Mac-Mini-64:~ Lisa$

**Reply by lsaffel** [reply]
brew config:
..............................................................
LV-Mac-Mini-64:~ Lisa$ `brew config`
HOMEBREW_VERSION: 3.1.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e0bfcdaf258dadaaf8c102d4cb51f23b6fdb2c4c
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.3.1-x86_64
CLT: 12.5.0.0.1.1617976050
Xcode: N/A
LV-Mac-Mini-64:~ Lisa$

**Reply by carlocab** [reply]
Your core tap is missing. Try
```
rm -rf $(brew --repo homebrew/core)
brew tap homebrew/core
```

**Reply by lsaffel** [reply]
ok, thanks. I executed those two commands. Now, I get this:
LV-Mac-Mini-64:~ Lisa$ `brew update`
Updated 1 tap (homebrew/core).
==> Updated Formulae
kustomize              wartremover            xrootd                 yaz                    zabbix
LV-Mac-Mini-64:~ Lisa$

**Reply by lsaffel** [reply]
but.... brew doctor still gives me lots:

LV-Mac-Mini-64:~ Lisa$ `brew doctor`
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8-config

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libpolar-3.6.0.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/node/cppgc/allocation.h
  /usr/local/include/node/cppgc/common.h
  /usr/local/include/node/cppgc/custom-space.h
...
... (( long list of files continue, here. Nothing after these now ))
...

**Reply by carlocab** [reply]
Is anything actually broken? The first thing `brew doctor` say is
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!
```

**Reply by lsaffel** [reply]
OK, I'm not sure. I just thought perhaps I shouldn't be getting all those warnings. I guess I will carry on and see what happens.

**Reply by linfulent** [reply]
Thanx

**Reply by developerbello** [reply]
aWESOME, THANKS MAN

**Reply by atbines** [reply]
awsome ,thanks 
solved my problem 👍

**Reply by Zuxxypml** [reply]
Awesome thia helped solved my error

**Reply by Kephass** [reply]
Thanks for this man.

**Reply by U1ltra** [reply]
This solve my similar problem super fast. Thanks

**Reply by sajad-alhussieny** [reply]
thanks a lot😊

**Reply by andidp** [reply]
thanks alot

**Reply by patghara** [reply]
solved my problem. Thanks a lot!

**Comment by Yonggie** [comment]
solved my problems!
thanks!
```bash
rm -rf $(brew --repo homebrew/core)
brew tap homebrew/core
```

**Reply by ShixiangWan** [reply]
Too TM thanks

**Reply by iitsmel** [reply]
Thank you!!

**Reply by denys-kohut** [reply]
Thank you!

**Reply by singhbrajesh13** [reply]
worked.

**Reply by xani-dev** [reply]
Worked perfectly, thanks!

**Reply by ahmdyasser** [reply]
dude, i love you

**Reply by yapdianhao** [reply]
Worked, thanks!

**Reply by knowself** [reply]
Yeah. I had ...

brew update 
fatal: Could not resolve HEAD to a revision

So, 

rm -rf $(brew --repo homebrew/core)
brew tap homebrew/core

Cured my problem.

**Reply by rekarromar** [reply]
Thankssss

**Reply by aneal-rta** [reply]
yes, this! thank you

**Reply by mickieann** [reply]
Fixed mine too! Tysm

**Reply by murius007** [reply]
Worked like a charm for me too. Thank you so much!

**Reply by mohamad22mk** [reply]
yes yes yes thanks thanks thanks

**Reply by jso-lab** [reply]
These commands saved me after a long time of research... Thanks a lot !

**Reply by ShinjiX-Web** [reply]
Solved my issue as well. Now installing Shopify CLI. Thanks!!!

**Reply by codepriezt** [reply]
This command  solved my issue after a long time of searching for a solution. thanks!

**Reply by Ruben-Tjok** [reply]
Thank you, worked like a charm

**Reply by ratio-dd** [reply]
Perfectly Fix My Problem！

**Reply by lsaffel** [reply]
I’m glad that me asking this question and a couple people responding seems to have helped lots of people! 😆

**Reply by itwazluck** [reply]
Thank you!

**Reply by devikachandnani** [reply]
this helped me! thank you so much!

**Reply by Sincere-io** [reply]
Wow, it really worked out, thank you very much！

**Reply by andremarquezz** [reply]
Thanks, my problem is solved, i am running on wsl2 on windows 11.

**Reply by shakeelbabar** [reply]
Thanks for you help

**Comment by amrangry** [comment]
After running  $ brew doctor 

it gives the following warning recommendation to slove 

$  rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Running the above solve my problem

**Comment by dyvopes** [comment]
Running the below solve my problem:
$ brew doctor
$ git -C $(brew --repo homebrew/core) checkout master

**Reply by xiaobing7** [reply]
Solved. Thank you!!!!

**Comment by chideraike** [comment]
For guys that  `pipenv install mysqlclient` didn't work :bug:, this was the solution that worked for me :white_check_mark:

 1. Update homebrew :package:
 - `rm -rf $(brew --repo homebrew/core)`
 - `brew tap homebrew/core`

 2. Then install MySQL :card_file_box:
 - `brew install mysql-connector-python`
 - `brew install mysql`

 3.  Finally :construction:
 - `pipenv install mysqlclient`

:tada: The mysqlclient package successfully installed!

**Reply by flxtagi** [reply]
thanks that was the solution I was looking for hours!

**Reply by Shubbair** [reply]
this was helpful , thanks

**Comment by maarous** [comment]
It worked for me also, thanks @carlocab

**Comment by Blackode10011** [comment]
I solved my problem as follows: I called the command $home brew doctor and it showed me the path of directory conflicts
Warning: Homebrew/homebrew-core was not tapped properly! Run:
  rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

I removed the conflict location, and I did an update with the command 
$home brew/core 
And then install the 
$ brew install cask 
To install the updates and other dependencies

**Comment by funmerc** [comment]
I came across this issue as well, the simple solution that worked for me was:
`brew tap --repair; brew update;`
